app/models/manageiq/providers/oracle/infra_manager.rb

```diff
@@ -0,0 +1,101 @@
+require 'uri'
+
+class ManageIQ::Providers::Oracle::InfraManager < ManageIQ::Providers::InfraManager
+  require_nested :RefreshWorker
+  require_nested :RefreshParser
+  require_nested :Host
+  require_nested :Refresher
+  require_nested :Template
+  require_nested :Vm
+
+  def self.ems_type
+    @ems_type ||= "oraclevm".freeze
+  end
+
+  def self.description
+    @description ||= "Oracle Virtualization Manager".freeze
+  end
+
+  def self.default_blacklisted_event_names
+    []
+  end
+
+  def supports_port?
+    true
+  end
+
+  def supported_auth_types
+    %w(default)
+  end
+
+  def supports_authentication?(authtype)
+    supported_auth_types.include?(authtype.to_s)
+  end
+
+  def self.raw_connect(server, port, username, password)
+    require 'fog/oracle'
+
+    url = URI::HTTPS.build(:host => server, :port => port).to_s
```

Comment:
It passes url string directly to Excon and `Excon.new` uses `URI.parse` to parse the url string. Passing URI object would not work.
```
require 'uri'
server = 'ff02::1'
port = 1234
u = URI::HTTPS.build(:host => server, :port => port)
URI.parse(u)
# URI::InvalidURIError: bad URI(is not URI?)
```